### DISCUSSION TITLE
Image attachment doesn't work

### DISCUSSION THREAD

**Discussion by unoexperto** [outline/outline#1574] [General] [answered]
Guys, could you please provide example of working settings for AWS ? I cannot understand what I'm doing wrong :( Mine are

```
docker run -d --name=outline --restart always \
    --network isolated \
    -e NODE_ENV=production \
    -e DEPLOYMENT=self \
    -e DEBUG=cache,presenters,events \
    -e SECRET_KEY=xxx \
    -e URL=https://xxx.zzz.com \
    -e PORT=8081 \
    -e DATABASE_URL=postgres://uuu:ppp@psql:5432/outlinedb \
    -e PGSSLMODE=disable \
    -e REDIS_URL=redis://redis:6379 \
    -e SLACK_KEY=ddd \
    -e SLACK_SECRET=vvvv \
    -e GOOGLE_CLIENT_ID=vvvv \
    -e GOOGLE_CLIENT_SECRET=xxxx \
    -e AWS_ACCESS_KEY_ID=www \
    -e AWS_SECRET_ACCESS_KEY=ffff \
    -e AWS_REGION=us-east-1 \
    -e AWS_S3_UPLOAD_BUCKET_URL=https://s3.amazonaws.com \
    -e AWS_S3_UPLOAD_BUCKET_NAME=xxxzzz \
    -e AWS_S3_ACL=public-read \
    -e AWS_S3_UPLOAD_MAX_SIZE=26214400 \
    -e AWS_S3_FORCE_PATH_STYLE=false \
    -e FORCE_HTTPS=false \
    -e SMTP_HOST=smtp-relay.gmail.com \
    -e SMTP_PORT=25 \
    -e SMTP_FROM_EMAIL=noreply@zzz.com \
    outlinewiki/outline:latest
```

This almost works. I see POST request to `https://s3.amazonaws.com/walkmindoutline` with image. AWS responds with `204 No Content` **and** link to uploaded file in `Location` header. But apparently it's never used so image doesn't appear in the document. Link that I see in log is https://s3.amazonaws.com/walkmindoutlineus/uploads%2Fcbdecf64-69c4-449f-a8bb-1d7c286ad49c%2F235a56a4-889b-45c9-a554-14ea246208a1%2Fimage.png

Interestingly if I change `AWS_S3_UPLOAD_BUCKET_URL` to `s3.amazonaws.com` Outline acts as if it uploaded the image and even displays placeholder for it in the document. But it's not pushed to the bucket.

Could you please advise what I'm missing ?

I tried suggestions from #1092

**Comment by tommoor** [comment, marked as answer]
Try removing `AWS_S3_FORCE_PATH_STYLE=false`, seems like you're disabling it – but with that generic bucket url you'd need to use the path-style routing.

Thanks for posting _all_ the details about the setup, very helpful.
